templates/terraform/resource.erb

```diff
@@ -234,20 +234,50 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
     }
 <%    end -%>
 <%    if object.async.is_a? Api::OpAsync -%>
-    <% if object.async.result.resource_inside_response and not object.identity.empty? -%>
+<%      if object.async.result.resource_inside_response and not object.identity.empty? -%>
     var response map[string]interface{}
     err = <%= client_name_camel -%>OperationWaitTimeWithResponse(
     config, res, &response, <% if has_project -%> project, <% end -%> "Creating <%= object.name -%>",
         int(d.Timeout(schema.TimeoutCreate).Minutes()))
-
     if err != nil {
-<%      if object.custom_code.post_create_failure -%>
+<%        if object.custom_code.post_create_failure -%>
         resource<%= resource_name -%>PostCreateFailure(d, meta)
-<%      end -%>
+<%        end -%>
         // The resource didn't actually create
         d.SetId("")
         return fmt.Errorf("Error waiting to create <%= object.name -%>: %s", err)
     }
+
+<%      if object.nested_query -%>
+    res, err = flattenNested<%= resource_name -%>(d, meta, res)
```

Comment:
Oh boy, I don't love having a `res` and a `response` floating around. This section of the code doesn't look like it actually does anything besides potentially remove the object from state, because the next part of it is still looking for the value in `response`. So for the generated acm_sp_resource code, it looks like it would still `d.Set("resource")` to the wrong value since it's looking for the value from the operation instead of from the flattener.

Reply:
yup! I realized after, please ignore review requests for now